src/devices/src/virtio/test_utils.rs

```diff
@@ -173,6 +173,17 @@ impl<'a> VirtqDesc<'a> {
             .is_ok());
     }
 
+    pub fn check_data_from_offset(&self, expected_data: &[u8], offset: usize) {
```

Comment:
I wouldn't add a separate method for this. I would just extend `check_data()` and provide `offset = 0` to all the old use cases.